src/app/modules/submission/encrypt-submission/encrypt-submission.utils.ts

```diff
@@ -237,3 +240,26 @@ export const createEncryptedSubmissionDto = (
     version: submissionData.version,
   }
 }
+
+export const getPaymentAmount = (
+  formPaymentFields: FormPaymentsField, // fields that are from document.form
+  incomingSubmissionPaymentFields?: PaymentFieldsDto, // fields that are from incoming submission
```

Comment:
It's checked in 
`incomingSubmissionPaymentFields?.amount_cents` or am I misunderstanding something?

Reply:
Hmm. with the current flow, when incomingSubmissionPaymentFields is empty, we will return 500 status code, but that should not be the case as this field comes from api request, right? In that case, we may need to do a separate validation for an early return with 400, please let me know if i misunderstanding anything 😅

Reply:
Only for `PaymentType.Variable` then we will need `incomingSubmissionPaymentFields` to be present

Reply:
I think it makes sense to be server-side error, since the user isn't able to fix this. Suggesting that this is highly likely a coding bug on the server side.